src/backend/distributed/commands/index.c

```diff
@@ -168,6 +168,69 @@ PlanIndexStmt(IndexStmt *createIndexStatement, const char *createIndexCommand)
 }
 
 
+/*
+ * ErrorIfReindexOnDistributedTable determines whether a given REINDEX
+ * involves a distributed table, & raises an error if so.
+ */
+void
+ErrorIfReindexOnDistributedTable(ReindexStmt *ReindexStatement)
+{
+	Relation relation = NULL;
+	Oid relationId = InvalidOid;
+	bool isDistributedRelation = false;
+	LOCKMODE lockmode = AccessShareLock;
+
+	/*
+	 * We first check whether a distributed relation is affected. For that, we need to
+	 * open the relation.
+	 */
+	if (ReindexStatement->relation == NULL)
+	{
+		/* ignore REINDEX SCHEMA, REINDEX SYSTEM, and REINDEX DATABASE */
+		return;
+	}
+
+	Assert(ReindexStatement->relkind == REINDEX_OBJECT_INDEX ||
```

Comment:
This fails in postgres 11. Make sure test with `--enable-cassert` before merging.